### PR TITLE
Fix for PHP 7.3

### PR DESCRIPTION
**Pull request by remicollet**

### CODE REVIEW

**Comment by remicollet** [comment]
@Jan-E please test last commit

File: rar_error.c [hunk 1]
```diff
@@ -252,7 +252,7 @@ void minit_rarerror(TSRMLS_D) /* {{{ */
 	rarexception_ce_ptr->ce_flags |= ZEND_ACC_FINAL;
 	zend_declare_property_bool(rarexception_ce_ptr, "usingExceptions",
 		sizeof("usingExceptions") -1, 0L /* FALSE */,
-		ZEND_ACC_PRIVATE | ZEND_ACC_STATIC TSRMLS_CC);
+		ZEND_ACC_STATIC TSRMLS_CC);
```
Comment (remicollet) [inline rar_error.c]: Cannot be private else cannot be retieve

`Fatal error: Uncaught Error: Cannot access private property RarException::$usingExceptions in /work/GIT/php-rar/tests/037.php:8`

File: rararch.c [hunk 2]
```diff
@@ -932,55 +932,6 @@ static void rararch_it_move_forward(zend_object_iterator *iter TSRMLS_DC);
...
-		it->empty_iterator = 1;
-	}
-	else
-		it->empty_iterator = 0;
-
-	_rar_entry_search_start(rar, RAR_SEARCH_TRAVERSE, &it->state TSRMLS_CC);
-	return (zend_object_iterator*) it;
-}
-/* }}} */
-
```
Comment (remicollet) [inline rararch.c]: Mostly only moved lower

**Comment by cmb69** [comment]
@Jan-E No, that doesn't appear to be documented. I wouldn't know what to document exactly, though.

**Comment by remicollet** [comment]
> No, that doesn't appear to be documented.

@cmb69 @Jan-E an-E is it ;)
https://github.com/php/php-src/commit/0c3561892b1f396620bee520f69bd904e2b76dc1

(ok, very recently)

**Comment by Jan-E** [comment]
> @Jan-E please test last commit

The complete PR worked fine.

**Comment by remicollet** [comment]
1 additional fix for 7.3.0beta
```
 z. HAVE_ST_BLKSIZE must be replaced with HAVE_STRUCT_STAT_ST_BLKSIZE and
    HAVE_ST_RDEV must be replaced with HAVE_STRUCT_STAT_ST_RDEV.

```

**Comment by alfredbez** [comment]
Any chance to get a new release? `4.0.0` is already 2.5 years old and does not work with PHP 7.3 and PHP 7.4. I installed the rar extension from this GitHub repository on Ubuntu 18.04 / PHP 7.4 and it worked very well, but I guess there are some folks who are trying to download the latest package from here: https://pecl.php.net/package/rar

```
abez@abez-XPS-13-9370:~$ php -v
PHP 7.4.1 (cli) (built: Dec 18 2019 14:44:22) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Xdebug v2.9.0, Copyright (c) 2002-2019, by Derick Rethans
    with Zend OPcache v7.4.1, Copyright (c), by Zend Technologies
abez@abez-XPS-13-9370:~$ php -m | grep rar
rar
abez@abez-XPS-13-9370:~$

```

**Comment by alfredbez** [comment]
I created a bug report for this: https://bugs.php.net/bug.php?id=79130

**Comment by Dibbyo456** [comment]
@alfredbez So in order to use it with PHP 7.3 and 7.4 we need to build from source from the master branch? If that is the case then installing on Linux would be easy but for Windows that's too much trouble.

**Comment by Jan-E** [comment]
I am building it regularly on Windows:
https://www.apachelounge.com/viewtopic.php?t=6359

It is up to you if you trust it or not.

**Pull request merged**